developerportal/templates/atoms/social-meta.html

```diff
@@ -1,31 +1,37 @@
+{% load static %}
 {% load wagtailcore_tags %}
 {% load wagtailimages_tags %}
 {% load app_tags %}
 
 <meta property="og:title" content="{{ page.title }}">
-{% if page.card_description %}
-  <meta name="description" content="{{ page.card_description }}">
-  <meta property="og:description" content="{{ page.card_description }}">
+{% with page.description|richtext|striptags as rich_description %}
+  {% firstof page.search_description page.card_description rich_description as description %}
+  {% if description %}
+    <meta name="description" content="{{ description }}">
+    <meta property="og:description" content="{{ description }}">
+  {% endif %}
+{% endwith %}
+
+{% if page.keywords.all %}
+  <meta name="keywords" content="{{ page.keywords.all|join:', ' }}">
 {% endif %}
+
 {% with self.get_site.site_name as site_name %}
   {% if site_name %}
     <meta property="og:site_name" content="{{ site_name }}">
   {% endif %}
 {% endwith %}
 
 {% get_scheme_and_host request as base_url %}
-{% if page.card_image or page.image %}
-  {% if page.card_image %}
-    {% image page.card_image width-1200 as image %}
-    <meta property="og:image" content="{{ base_url }}{{ image.url }}">
-    <meta property="og:image:width" content="{{ image.width }}">
-    <meta property="og:image:height" content="{{ image.height }}">
-  {% else %}
-    {% image page.image width-1200 as image %}
-    <meta property="og:image" content="{{ base_url }}{{ image.url }}">
-    <meta property="og:image:width" content="{{ image.width }}">
-    <meta property="og:image:height" content="{{ image.height }}">
-  {% endif %}
+{% static "img/placeholders/article.jpg" as fallback_image_url %}
+{% if page.card_image %}
+  {% image page.card_image width-1200 as image %}
+{% elif page.image %}
+  {% image page.image width-1200 as image %}
 {% endif %}
+<meta property="og:image" content="{{ base_url }}{% if image %}{{ image.url }}{% else %}{{ fallback_image_url }}{% endif %}">
+<meta property="og:image:width" content="{% if image %}{{ image.width }}{% else %}960{% endif %}">
```

Comment:
Super nitpick: if the article.jpg placeholder is recut or replaced at any point, these would become invalid. Maybe defining them along with the `fallback_image_url` a few lines above would at least put all the things that might need to be tweaked adjacent to each other. 

See what I mean about super-nit-picky? 😄 Feel free to skip if you disagree

Reply:
I've added some variables for the width/height with `{% with %}`. Not sure if this is what you were meaning, but let me know!